snappy/kernel_os.go

```diff
@@ -68,6 +68,15 @@ func extractKernelAssets(s *snap.Info, snapf snap.File, flags InstallFlags, inte
 		return fmt.Errorf("can not extract kernel assets from snap type %q", s.Type)
 	}
 
+	// sanity check that we have the new kernel format
+	kmeta, err := snapf.ReadFile("meta/kernel.yaml")
```

Comment:
from the point of view a snap.ReadKernelInfo() taking a Info or PlaceInfo and doing validation may make sense